lib/ansible/modules/git.py

```diff
@@ -1293,6 +1318,16 @@ def main():
     # iface changes so need it to make decisions
     git_version_used = git_version(git_path, module)
 
+    if key_file_content:
+        key_file_content += "\n" if not key_file_content.endswith("\n") else ""
+        key = to_bytes(key_file_content)
+        # tempfile not saved on disk
+        tmpfd, key_file = tempfile.mkstemp()
+        module.add_cleanup_file(key_file)
+        tmpfile = os.fdopen(tmpfd, "w+b")
+        tmpfile.write(key)
```

Comment:
We have avoided until now being responsible for writing/deleting private data to disk, this introduces a fail point that we normally want to avoid.

I'm not sure core want's to accept this feature at this time, we are still discussing this so this is not an outright rejection.

Reply:
Thank you @bcoca. Feel free to close PR (depending on the discussion's outcome)